easybuild/easyblocks/o/openmpi.py

```diff
@@ -59,41 +60,69 @@ def config_opt_unused(key, enable_opt=False):
-        for dep in ['CUDA', 'hwloc', 'libevent', 'libfabric', 'PMIx', 'UCX']:
-            if config_opt_unused(dep.lower()):
-                dep_root = get_software_root(dep)
-                if dep_root:
-                    if dep == 'libfabric':
-                        opt_name = 'ofi'
-                    else:
-                        opt_name = dep.lower()
-                    self.cfg.update('configopts', '--with-%s=%s' % (opt_name, dep_root))
+        for dep in known_dependencies:
+            opt_name = dep.lower()
+            # If the option is already used, don't add it
+            if config_opt_used(opt_name):
+                continue
+
+            # libfabric option renamed in OpenMPI 3.1.0 to ofi
+            if dep == 'libfabric' and LooseVersion(self.version) >= LooseVersion('3.1'):
+                opt_name = 'ofi'
+                # Check new option name. They are synonyms
+                if config_opt_used(opt_name):
+                    continue
+
+            dep_root = get_software_root(dep)
+            # If the dependency is loaded, specify its path, else use the "unused" value, if any
+            if dep_root:
+                opt_value = dep_root
+            else:
+                opt_value = unused_dep_value.get(dep)
+            if opt_value is not None:
+                self.cfg.update('configopts', '--with-%s=%s' % (opt_name, opt_value))
+
+        if bool(get_software_root('PMIx')) != bool(get_software_root('libevent')):
+            raise EasyBuildError('You must either use both PMIx and libevent as dependencies or none of them. '
+                                 'This is to enforce the same libevent is used for OpenMPI as for PMIx or '
+                                 'the behavior may be unpredictable.')
 
         # check whether VERBS support should be enabled
-        if config_opt_unused('verbs'):
+        if not config_opt_used('verbs'):
 
             # for OpenMPI v4.x, the openib BTL should be disabled when UCX is used;
             # this is required to avoid "error initializing an OpenFabrics device" warnings,
             # see also https://www.open-mpi.org/faq/?category=all#ofa-device-error
             if LooseVersion(self.version) >= LooseVersion('4.0.0') and '--with-ucx' in self.cfg['configopts']:
-                self.cfg.update('configopts', '--without-verbs')
-
+                verbs = False
             else:
                 # auto-detect based on available OS packages
-                verbs = False
-                for osdep in ['libibverbs-dev', 'libibverbs-devel', 'rdma-core-devel']:
-                    if check_os_dependency(osdep):
-                        verbs = True
-                        break
-
-                if verbs:
-                    self.cfg.update('configopts', '--with-verbs')
-                else:
-                    self.cfg.update('configopts', '--without-verbs')
+                os_packages = ('libibverbs-dev', 'libibverbs-devel', 'rdma-core-devel')
```

Comment:
For the sake of having only one place to update, this should be taken from the OS_PKG_IBVERBS_DEV in EASYCONFIG_CONSTANTS from framework/easyconfig/constants.py

Reply:
Done